code/modules/food/recipe.dm

```diff
@@ -0,0 +1,321 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * *
+ * /datum/recipe by rastaf0            13 apr 2011 *
+ * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * This is powerful and flexible recipe system.
+ * It exists not only for food.
+ * supports both reagents and objects as prerequisites.
+ * In order to use this system you have to define a deriative from /datum/recipe
+ * * reagents are reagents. Acid, milc, booze, etc.
+ * * items are objects. Fruits, tools, circuit boards.
+ * * result is type to create as new object
+ * * time is optional parameter, you shall use in in your machine,
+     default /datum/recipe/ procs does not rely on this parameter.
+ *
+ *  Functions you need:
+ *  /datum/recipe/proc/make(var/obj/container as obj)
+ *    Creates result inside container,
+ *    deletes prerequisite reagents,
+ *    transfers reagents from prerequisite objects,
+ *    deletes all prerequisite objects (even not needed for recipe at the moment).
+ *
+ *  /proc/select_recipe(list/datum/recipe/available_recipes, obj/obj as obj, exact = 1)
+ *    Wonderful function that select suitable recipe for you.
+ *    obj is a machine (or magik hat) with prerequisites,
+ *    exact = 0 forces algorithm to ignore superfluous stuff.
+ *
+ *
+ *  Functions you do not need to call directly but could:
+ *  /datum/recipe/proc/check_reagents(var/datum/reagents/avail_reagents)
+ *  /datum/recipe/proc/check_items(var/obj/container as obj)
+ *
+ * */
+ 
+// Recipe type defines. Used to determine what machine makes them. TODO: Add Grill to the list.
+#define MICROWAVE			0x1
+#define FRYER				0x2
+#define OVEN				0x4
+#define CANDYMAKER			0x8
+#define CEREALMAKER			0x10
+
+/datum/recipe

```

Comment:
IIRC using generic /recipe datum rather than microwave-specific subtype may cause troubles in longer run without refactoring Particle Smasher, which once again IIRC, uses subtypes of this here datum for it's own recipies.

Reply:
You could probably better refactor the particle smasher to be a subtype of appliance to use some of the stuff, but... when tf was the last time anyone used a particle smasher? I've never seen it used, but if it's deemed necessary, I can look at reworking it.

Reply:
Issue with Particle Smasher last I heard wasn't that it itself broke but that it in reverse broke microwaves (making some of recipies output a steel sheet for example). I do believe that issue was fixed, but overall situation with recipe datums, afaik, never changed, so that'd need looking into it. Plus, fact that Particle Smasher's only commonality with kitchen appliance is that 'it also can have items inserted and also transforms them', while having nothing to do with food, cooking, etc, never made sense to use same parent of a recipe.

Reply:
So what I'm hearing is we need to separate Particle Smasher entirely from the recipe system/cooking in general, lol.

Reply:
It shouldn't be too hard, just changing base subtype to  be it's own thing and moving over only relevant vars would do the trick. I know Polaris wanted to unionize all recipies for everything to be more of generic system... But until they do it's best to keep food recipies as one thing and advanced ultra material synthesis as another.

Also

Reply:
So taking a look at the particle smasher, it uses /datum/particle_smasher_recipe, so that shouldn't conflict here at all, I'm going to mark this resolved if there's no other concerns.

Reply:
Aha, so they did finally change that, probablt just missed the PR. Yeah, mark it down, that shouldn't be an issue anymore. Doesn't make it any less dumb that it happened in first place, but whatevs.